js/mage/adminhtml/wysiwyg/tinymce/setup.js

```diff
@@ -70,6 +70,11 @@
         var settings = {
             selector: this.selector,
             config: this.config,
+            valid_children: '+body[style]',
+            custom_elements:"style,~style",
+            protect: [
+                /([\S]?)<script([\s\S]*?)>([\S]*?)<\/script([\s]*?)>([\S]?)/ig
```

Comment:
👍 it's ok acceptable

Reply:
what's the difference between \s and \S<'

Reply:
![explain](https://github.com/OpenMage/magento-lts/assets/5071467/f204e18d-0a71-4ee9-9afa-9b847fd68cdf)

i test it several cases here:
https://regex101.com/r/Ub4FeY/1

Reply:
Bit shorter, w/o grouping ... ?

`/<script[\s\S]*?>[\s\S]*?<\/script>/gi`

---

Do we want script tags in cms blocks/pages?

Reply:
> 👍 it's ok acceptable

I dont unterstand it. Why is it acceptable?

Reply:
I understand all the obvious observations, but the regex needs to be tested. It is the result of many trials

Reply:
> Do we want script tags in cms blocks/pages?
Yes, this PR is to support script tags in CMS blocks/pages content, similar to how many themes, like Porto, do.
Maybe we can make this configurable and let tinyMCE automatically strips script tags but must be allowed by default.
With this change, script tags will be encrypted in the TinyMCE editor and not executed.

Reply:
> Do we want script tags in cms blocks/pages?

Yes, this PR is to support script tags in CMS blocks/pages content, similar to how many themes, like Porto, do.
Maybe we can make this configurable and let tinyMCE automatically strips script tags but must be allowed by default.
With this change, script tags will be encrypted in the TinyMCE editor and not executed.

Reply:
Script tag was only a question, but CodeQL should show no regex errors.

Reply:
now catch also </script foo="bar"> and is codeql compliant
https://regex101.com/r/Ub4FeY/2
Even though I don't see a real security risk, the regexp has to encrypt the scripts in TinyMCE to prevent their deletion. However, they remain on frontend